coalib/coala.py

```diff
@@ -59,11 +60,34 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
+                get_all_bears)
+            filtered_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead.")
+                args.filter_by = [] if args.filter_by is None else \
+                    args.filter_by
+                args.filter_by.append(['language'] + args.filter_by_language)
+            if args.filter_by:
+                # Each iteration of the following loop applies
+                # filters one by one provided as arguments
+                for filter_by in args.filter_by:
+                    filter_by_x, *filter_args = filter_by
+                    try:
+                        filtered_bears = FilterHelper.get_filtered_bears(
+                            filter_by_x, filter_args, filtered_bears)
+                    except NotImplementedError:
+                        logging.exception(filter_by_x +
+                                          ' filter is not implemented.')
+                        return 255
+                    except AttributeError as ex:
+                        console_printer.print(ex)
```

Comment:
ahhh you raise an `AttributeError` for non-existing filters. I would consider implementing a new exception, like `InvalidFilterException`. An attribute error is so general that it might occur everywhere, and possible bugs would slip through.